frontend/src/concepts/k8s/pods/utils.ts

```diff
@@ -57,21 +68,33 @@ ${logsIndividualStep}`,
 };
 
 export const getPodStepsStates = async (
-  podContainers: PodContainer[],
+  podContainerStatuses: (PodContainerStatus | undefined)[],
```

Comment:
I already see you made this type `PodContainerStatuses` in the type file, maybe reuse it?